src/app-layer-ssl.c

```diff
@@ -1553,8 +1553,7 @@ static int TLSDecodeHandshakeHello(SSLState *ssl_state,
     /* Ensure that we have a JA4 state defined by now, we are in a client/server hello
        and we don't have such a state yet (to avoid leaking memory in case this function
        is entered more than once). */
-    if (SC_ATOMIC_GET(ssl_config.enable_ja4) &&
-            ssl_state->current_flags &
+    if (ssl_state->current_flags &
```

Comment:
@jasonish what do you think about this commit ?

Reply:
Seems the `if` is still relevant, as we're testing for it as part of the if statement?

Reply:
I mean it removes the config check `ssl_config.enable_ja4` for ja4 computation, and moves it just before logging it...
Does that look ok ?

Reply:
I think the original intention was to avoid computing it when disabled, not simply avoid logging it. Something else that I'm realizing is just how much JA4 is still in the code with the `--disable-ja4` compile time option. I wonder how this plays with that.

Reply:
Yeah, that's the implication of the change. So in order to 'track' the JA4 object in order to be able to extract the information without explicitly enabling the fingerprint, we must do this. But as you rightly say, this makes suricata keep track of the objects fields. Although, it doesn't make it compute the hash! Let's be clear! So its wether we want to force the user to ``ja4-fingerprints=yes`` in order to say extract the handshake information only (i.e. without the actual fingerprint).

Reply:
Would it make sense to disassociate that fields that go into computing JA4, from JA4 itself. That way they could be tracked and logged even if ja4 is disabled at compile time? Thoughts @victorjulien @catenacyber ?

Reply:
I find that a little confusing, sorry! So the fields that are used to make up a JA4 are to be unrelated to a JA4? Or are you meaning for it to be a handshake object that can produce a JA4? With this change, it really is just that, only it's called JA4. Maybe calling it `Handshake` would be more suited? So they are tracked without JA$ enablement but we only produce the hash when required? Really, that is what I have produced. So I'm a little uncertain of what you're meaning..

Reply:
@victorjulien @jasonish @catenacyber hey guys! Can I get a response on this, if you get some time, please? I'd like to get this completed asap as I'm completing work that relies on this output. I propose we remove the ability to extract the fields without explicitly enabling `ja4-fingerprint=yes`- so effectively just removing my last commit! The reason is the scope of this work was originally intended to only produce the 'raw fields' of the JA4 object and I think anything else is extending this. Would you agree?

Reply:
@jasonish what would it mean practically if we follow your suggestion? What kind of internal tracking changes and output changes would it involve?

Reply:
> @jasonish what would it mean practically if we follow your suggestion? What kind of internal tracking changes and output changes would it involve?

I don't know myself. Just thinking high level here.  The ticket states:

> My goal is to be able to reproduce the JA4 hash outside of suricata

Would it be desirable to do that with JA4 compiled out stil?